### ISSUE TITLE
Edit page link not working behind reverse proxy

### ISSUE THREAD

**Issue by djhworld** [ryanlelek/Raneto#310]
I'm running Raneto behind a reverse proxy.

This _mostly_ works fine in terms of rendering notes properly, except when you open a note (e.g. `data.md`), and select **Actions -> Edit** you get directed to `https://mydomain/data/edit` instead of `https://mydomain/notes/data/edit` because it seems to ignore the `base_url` property I have configured

I've had a look at the template for the Edit link and it relies on this `current_url` property (https://github.com/gilbitron/Raneto/blob/master/themes/default/templates/page.html#L59) in https://github.com/gilbitron/Raneto/blob/master/app/routes/wildcard.route.js#L105

If I update this to 

```
current_url   : req.protocol + '://' + req.get('host') + config.base_url + req.originalUrl,
```

It fixes the problem, but I'm not sure if that's the right way to go about it seeing as people might put FQDNs in the `base_url` 

**Setup I'm running**

* latest version of https://github.com/linuxserver/docker-raneto/releases image
* Caddy reverse proxy

Relevant caddy section:

```
	proxy /notes pi3:8922 {
	    transparent
	    without /notes
	}
```

raneto config.js

```
base_url: '/notes'
```

**Comment by Synt3x** [comment]
Latest version of Raneto from linuxserver.io works great for me with Traefik 1.7 as a reverse proxy.

**Comment by ryanlelek** [comment]
Ok, your suggestion tested out fine in my quick checks.
I did change `base_url` from your suggestion to `path_prefix`
`base_url` is intended to be the FQDN as you pointed out.
There is now a `path_prefix` config variable to implement your fix.

And really, we should probably be using `base_url` to populate `current_url` variable, instead of assembling in-line.
Lots of PRs have been submitted (which is a great thing) and some cleanup is needed for consistency.

Here are the values I used in config.js:
```
base_url: 'http://local.raneto.com/raneto',
path_prefix: '/raneto',
```

Another important thing to note is Caddy is removing the path prefix, before proxying.
Not everyone may be removing the prefix, and we might need to support that as well.

For those that find this, here is the Nginx config used in testing for reference, because removing the path is complicated.

```
location ~ ^/raneto(/?)(.*) {
        proxy_pass        http://raneto/$2;
        proxy_redirect    off;
        proxy_set_header  Host $host;
        proxy_set_header  X-Real-IP $remote_addr;
        proxy_set_header  X-Forwarded-For $proxy_add_x_forwarded_for;
}

upstream raneto {
    server 127.0.0.1:3000 max_fails=10 fail_timeout=4s;
}
```
